### DISCUSSION TITLE
Version 0.24

### DISCUSSION THREAD

**Discussion by lovelydinosaur** [encode/httpx#2534] [General]
We've got a few upcoming behavioural changes, that I think we ought to push on with now.
This will commit us to our next release being version 0.24.

Here's what I think we'd like to 

* Fix `iter_lines()` to match the behaviour of `splitlines()`. No trailing newlines on strings.
  Pull request https://github.com/encode/httpx/pull/2423
  Would close #2422.

* Add explicit `NetRCAuth()` class instead of built-in client behaviour.
  Pull request https://github.com/encode/httpx/pull/2535
  Would close #2532.
  ~~Would close #2088~~. *edit: no, it wouldn't*

* Remove our `rfc3986` dependancy.
  Pull request #2252.
  Would close #1833
  Would close #2169.

* Prefer `%20` over `+` for spaces in query parameters.
  Brings us more in line with browser behaviours.
  Pull request #2543.
  Would close #2536.
  See discussion #2460.

* Improve logging, adding `httpcore` logging, and switching to a more standard config style.
  Pull request #2547
  Pull request https://github.com/encode/httpcore/pull/648
  Would close https://github.com/encode/httpx/issues/2042.

* I also intend to raise a discussion around making `idna` support be opt-in, rather than on-by-default, which might or might not be worth aiming at 0.24.

Are we all good with pushing on with getting a 0.24 release on track?

**Comment by paulschreiber** [comment]
What's the expected release date for 0.24? This will stop dependabot from trying to push an update which causes a dependency conflict with `rfc3986`.

**Comment by ZupoLlask** [comment]
Please release that version ASAP. Thanks.

**Comment by ZupoLlask** [comment]
@tomchristie Can you please have a look at this? Thanks

**Reply by zanieb** [reply]
https://github.com/encode/httpx/pull/2547 is being finalized and is the last remaining item in the list Tom wrote above.

**Reply by ydennisy** [reply]
@madkinsz are we close?

**Reply by ZupoLlask** [reply]
We are: https://github.com/encode/httpx/pull/2652

**Comment by ericof** [comment]
@tomchristie, @madkinsz: Is there any blocker for a 0.24 release? Is there anything we can help?
I've been using httpx from HEAD, which is less than ideal, for a while and would love to go back to installing from pypi.

**Comment by ZupoLlask** [comment]
I guess we'll see a new release in the next few hours: https://github.com/encode/httpx/pull/2652

Thanks team!
